tests/cloner_test.go

```diff
@@ -574,6 +575,75 @@ var _ = Describe("all clone tests", func() {
 						string(dv.Status.Progress) == "N/A"
 				}, timeout, pollingInterval).Should(BeTrue(), "DV Should succeed with storage.prePopulated==pvcName")
 			})
+
+			DescribeTable("Should clone with empty volume size without using size-detection pod",
+				func(sourceVolumeMode v1.PersistentVolumeMode, targetVolumeMode v1.PersistentVolumeMode) {
+					// When cloning without defining the target's storage size, the source's size can be attainable
+					// by different means depending on the clone type and the volume mode used.
+					// Either if "block" is used as volume mode or smart/csi cloning is used as clone strategy,
+					// the value is simply extracted from the original PVC's spec.
+					if cloneType == "network" && sourceVolumeMode == v1.PersistentVolumeFilesystem {
+						Skip("Clone strategy and volume mode combination requires of size-detection pod")
+					}
+
+					if sourceVolumeMode != targetVolumeMode && cloneType == "csivolumeclone" {
+						Skip("csivolumeclone only works for the same volumeMode")
+					}
+
+					if (sourceVolumeMode == v1.PersistentVolumeBlock || targetVolumeMode == v1.PersistentVolumeBlock) && !f.IsBlockVolumeStorageClassAvailable() {
+						Skip("Storage Class for block volume is not available")
+					}
+
+					dataVolume := utils.NewDataVolumeWithHTTPImportAndStorageSpec(dataVolumeName, "200Mi", fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+					dataVolume.Spec.Storage.VolumeMode = &sourceVolumeMode
+					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+					Expect(err).ToNot(HaveOccurred())
+					f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+					sourcePvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
```

Comment:
I think we should probably wait for the source DV to be completed before we attempt to clone it without a size. Although it is probably a good test to immediately attempt to clone it and make sure the clone eventually completes once the source is done.